SSL errors when using `gatsby develop --https`

##  Description

When running `gatsby develop --https` both with without `--cert-file`/`--key-file` SSL does not work in Firefox, Chrome or Safari (screenshots below).

### Steps to reproduce

1. I've created a minimal reproduction in https://github.com/patrickarlt/gatsby-ssl-issue-reproduction. This is based off the default starter with the latest version of Gatsby.
2. `npm install`
3. `npm run start:devcert` this starts the local dev server and automatically creates SSL certs.
4. Observe failures.
5. Install [`mkcert`](https://github.com/FiloSottile/mkcert), for Mac this means:
   * `brew install mkcert`
   * `brew install nss`
   * `mkcert install`
   * `mkcert localhost`
6. `npm run start:local-certs` this starts the dev server with the local certs from `mkcert`
7. Observe failures.

In order to check that the certs are valid you can use [`http-server`](https://www.npmjs.com/package/http-server):

* `npm i -g http-server`
* `http-server . -C localhost.pem -K localhost-key.pem -S -a localhost`

This displays the directory structure of our project tin the browser over `https://...` note the lock icon in Chrome:

![2020-05-20_12-26-26 (1)](https://user-images.githubusercontent.com/378557/82488677-25034f80-9a95-11ea-8d45-e2d536aed5b5.png)

### Expected result

The server should successfully serve pages over `https://...`.

### Actual result

Browsers produce SSL errors.

![2020-05-20_12-18-35](https://user-images.githubusercontent.com/378557/82488072-2c762900-9a94-11ea-8626-824ebde26e76.png)
![2020-05-20_12-19-02](https://user-images.githubusercontent.com/378557/82488086-2ed88300-9a94-11ea-975b-fd3dfa530a3e.png)
![2020-05-20_12-19-20](https://user-images.githubusercontent.com/378557/82488090-30a24680-9a94-11ea-94ea-784645cd3af8.png)

### Environment

**Gatsby**

```
  System:
    OS: macOS Mojave 10.14.6
    CPU: (8) x64 Intel(R) Core(TM) i7-6820HQ CPU @ 2.70GHz
    Shell: 5.3 - /bin/zsh
  Binaries:
    Node: 14.3.0 - ~/.nvm/versions/node/v14.3.0/bin/node
    npm: 6.14.5 - ~/.nvm/versions/node/v14.3.0/bin/npm
  Languages:
    Python: 2.7.10 - /usr/bin/python
  Browsers:
    Chrome: 81.0.4044.138
    Firefox: 77.0
    Safari: 13.1
  npmPackages:
    gatsby: ^2.22.3 => 2.22.3
```

**OpenSSL**

```
$ openssl version
LibreSSL 2.6.5
```

**mkcert**

```
$ brew info mkcert 
mkcert: stable 1.4.1 (bottled)
Simple tool to make locally trusted development certificates
https://github.com/FiloSottile/mkcert
/usr/local/Cellar/mkcert/1.4.1 (6 files, 5.3MB) *
  Poured from bottle on 2020-04-08 at 18:45:14
From: https://github.com/Homebrew/homebrew-core/blob/master/Formula/mkcert.rb
==> Dependencies
Build: go ✘
==> Analytics
install: 3,827 (30 days), 11,641 (90 days), 47,679 (365 days)
install-on-request: 3,351 (30 days), 10,186 (90 days), 40,390 (365 days)
build-error: 0 (30 days)
```

**nss**

```
$ brew info nss
nss: stable 3.52 (bottled)
Libraries for security-enabled client and server applications
https://developer.mozilla.org/docs/NSS
/usr/local/Cellar/nss/3.52 (225 files, 34.1MB) *
  Poured from bottle on 2020-05-19 at 16:52:58
From: https://github.com/Homebrew/homebrew-core/blob/master/Formula/nss.rb
==> Dependencies
Required: nspr ✔
==> Analytics
install: 38,963 (30 days), 105,774 (90 days), 394,028 (365 days)
install-on-request: 8,110 (30 days), 19,022 (90 days), 62,334 (365 days)
build-error: 0 (30 days)
```

Version 2.22.9 still failing tests. We are using cypress tests with https in our ci/cd, and dependabot tracking new versions of npm packages fails ci/cd tests with error: 

```
success createPages - 0.082s
success createPagesStatefully - 0.037s
success onPreExtractQueries - 0.001s
success update schema - 0.028s
success extract queries from components - 0.381s
success write out requires - 0.033s
success write out redirect data - 0.008s
success onPostBootstrap - 0.017s
⠀
info bootstrap finished - 44.105s
⠀
success run queries - 0.163s - 37/37 227.51/s
warning 
  When setting `useBuiltIns: 'usage'`, polyfills are automatically imported when needed.
  Please remove the `import '@babel/polyfill'` call or use `useBuiltIns: 'entry'` instead.
⠀
You can now view pooltickets-gatsby in the browser.
⠀
  https://localhost:8000/
⠀
View GraphiQL, an in-browser IDE, to explore your site's data and schema
⠀
  https://localhost:8000/___graphql

Note that the development build is not optimized.
To create a production build, use gatsby build
⠀
success Building development bundle - 30.032s
Error: Timeout
    at MergeMapSubscriber.project (/home/runner/work/pooltickets-gatsby/pooltickets-gatsby/node_modules/wait-on/lib/wait-on.js:145:74)
    at MergeMapSubscriber._tryNext (/home/runner/work/pooltickets-gatsby/pooltickets-gatsby/node_modules/rxjs/internal/operators/mergeMap.js:69:27)
    at MergeMapSubscriber._next (/home/runner/work/pooltickets-gatsby/pooltickets-gatsby/node_modules/rxjs/internal/operators/mergeMap.js:59:18)
    at MergeMapSubscriber.Subscriber.next (/home/runner/work/pooltickets-gatsby/pooltickets-gatsby/node_modules/rxjs/internal/Subscriber.js:66:18)
    at AsyncAction.dispatch [as work] (/home/runner/work/pooltickets-gatsby/pooltickets-gatsby/node_modules/rxjs/internal/observable/timer.js:31:16)
    at AsyncAction._execute (/home/runner/work/pooltickets-gatsby/pooltickets-gatsby/node_modules/rxjs/internal/scheduler/AsyncAction.js:71:18)
    at AsyncAction.execute (/home/runner/work/pooltickets-gatsby/pooltickets-gatsby/node_modules/rxjs/internal/scheduler/AsyncAction.js:59:26)
    at AsyncScheduler.flush (/home/runner/work/pooltickets-gatsby/pooltickets-gatsby/node_modules/rxjs/internal/scheduler/AsyncScheduler.js:52:32)
    at listOnTimeout (internal/timers.js:549:17)
    at processTimers (internal/timers.js:492:7)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
##[error]Process completed with exit code 1.
```

--- a/packages/gatsby/src/commands/develop.ts
+++ b/packages/gatsby/src/commands/develop.ts
@@ -116,24 +116,6 @@ module.exports = async (program: IProgram): Promise<void> => {
     getRandomPort(),
   ])
 
-  // NOTE(@mxstbr): We need to start the develop proxy before the develop process to ensure
-  // codesandbox detects the right port to expose by default
-  const proxy = startDevelopProxy({
-    proxyPort: proxyPort,
-    targetPort: developPort,
-    program,
-  })
-
-  const developProcess = new ControllableScript(`
-    const cmd = require(${JSON.stringify(developProcessPath)});
-    const args = ${JSON.stringify({
-      ...program,
-      port: developPort,
-      proxyPort,
-    })};
-    cmd(args);
-  `)
-
   // In order to enable custom ssl, --cert-file --key-file and -https flags must all be
   // used together
   if ((program[`cert-file`] || program[`key-file`]) && !program.https) {
@@ -167,6 +149,26 @@ module.exports = async (program: IProgram): Promise<void> => {
       directory: program.directory,
     })
   }
+
+  // NOTE(@mxstbr): We need to start the develop proxy before the develop process to ensure
+  // codesandbox detects the right port to expose by default
+  const proxy = startDevelopProxy({
+    proxyPort: proxyPort,
+    targetPort: developPort,
+    program,
+  })
+
+  const developProcess = new ControllableScript(`
+    const cmd = require(${JSON.stringify(developProcessPath)});
+    const args = ${JSON.stringify({
+      ...program,
+      port: developPort,
+      proxyPort,
+      // Don't pass SSL options down to the develop process, it should always use HTTP
+      ssl: null,
+    })};
+    cmd(args);
+  `)
 
   let unlock
   if (!isCI()) {

